### PR TITLE
fix: State Export Manually Executed Upgrade Path Testing Pipeline Merge.

### PR DESCRIPTION
**Pull request by gzukel**
# Description

This PR merges a pipeline that does upgrade path testing by standing up a network with massaged mainnet state and upgrading to the latest binary either by simply doing a binary replacement or governance upgrade if its a major version change. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

This was tested in Github Actions pipeline 6 times with success.

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [x] Tested via GitHub Actions 

# Checklist:

- [x] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -245,7 +245,7 @@ jobs:
         shell: alpine.sh --root {0}
         run: |
           git config --global --add safe.directory '*'
-          go mod tidy
+          #go mod tidy
```
Comment (CharlieMc0) [inline .github/workflows/build.yml:248]: Please delete the whole line since go mod tidy no longer works with go 1.20

**Review by billyyang423 — APPROVED** [review]

File: .github/workflows/upgrade_path_testing.yaml [hunk 2]
```diff
@@ -97,9 +141,7 @@ jobs:
           cat $DAEMON_HOME/config/genesis.json | jq '.app_state["mint"]["params"]["mint_denom"]="azeta"' > $DAEMON_HOME/config/tmp_genesis.json && mv $DAEMON_HOME/config/tmp_genesis.json $DAEMON_HOME/config/genesis.json
           cat $DAEMON_HOME/config/genesis.json | jq '.app_state["evm"]["params"]["evm_denom"]="azeta"' > $DAEMON_HOME/config/tmp_genesis.json && mv $DAEMON_HOME/config/tmp_genesis.json $DAEMON_HOME/config/genesis.json
           cat $DAEMON_HOME/config/genesis.json | jq '.consensus_params["block"]["max_gas"]="10000000"' > $DAEMON_HOME/config/tmp_genesis.json && mv $DAEMON_HOME/config/tmp_genesis.json $DAEMON_HOME/config/genesis.json
-          
-          contents="$(jq '.app_state.gov.voting_params.voting_period = "10s"' $DAEMON_HOME/config/genesis.json)" && \
-          echo "${contents}" > $DAEMON_HOME/config/genesis.json
+          cat $DAEMON_HOME/config/genesis.json | jq '.app_state["gov"]["voting_params"]["voting_period"]="60s"' > $DAEMON_HOME/config/tmp_genesis.json && mv $DAEMON_HOME/config/tmp_genesis.json $DAEMON_HOME/config/genesis.json
```
Comment (CharlieMc0) [inline .github/workflows/upgrade_path_testing.yaml:144]: Why not complete lines 137-144 in the python script along with the other genesis file modifications?

File: .github/workflows/upgrade_path_testing.yaml [hunk 3]
```diff
@@ -146,8 +195,215 @@ jobs:
           else:
               git_env_file.write("UPGRADE_TYPE=GOV")
           git_env_file.close()
- 
-      - name: "CHECK_UPGRADE_TYPE"
-        shell: shell
+
+      - name: "NON_CONCENSUS_BREAKING_UPGRADE"
```
Comment (CharlieMc0) [inline .github/workflows/upgrade_path_testing.yaml:199]: Typo `CONSENSUS` not `CONCENSUS`

File: .github/workflows/upgrade_path_testing.yaml [hunk 4]
```diff
@@ -146,8 +195,215 @@ jobs:
...
+            --gas-adjustment=2 \
+            --gas-prices=${{ env.GAS_PRICES }} \
+            -y
+          
+          sleep 5
+          zetacored query gov proposal ${proposal_id} --node http://127.0.0.1:26657
+          
+          TARGET_HEIGHT=$(echo ${UPGRADE_HEIGHT} | cut -d '.' -f 1)
+          ENDPOINT="http://127.0.0.1:26657"
+          MAX_TRIES=100
```
Comment (CharlieMc0) [inline .github/workflows/upgrade_path_testing.yaml:284]: Can you set these and env variables at the job level? Just so it's easier to see what values can be tweaked for fine tuning

File: .github/workflows/upgrade_path_testing.yaml [hunk 5]
```diff
@@ -146,8 +195,215 @@ jobs:
...
+          
+              # If we're on the last iteration and the block height hasn't changed for all tries
+              if [[ $i -eq $MAX_TRIES && $stalled_count -eq $MAX_TRIES ]]; then
+                  cat cosmovisor.log
+                  echo "Block height hasn't changed for $MAX_TRIES consecutive checks. Network might be stalled."
+                  exit 2
+              fi
+          
+              # Sleep for the specified duration
+              sleep $SLEEP_DURATION
```
Comment (CharlieMc0) [inline .github/workflows/upgrade_path_testing.yaml:408]: Why is there a sleep at the end when there is nothing left to wait for?

**Review by CharlieMc0 — CHANGES_REQUESTED** [review]

File: .github/actions/upgrade-testing/scripts/create_genesis.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    if account["address"] == "0x0000000000000000000000000000000000000001":
+        print("pop account", account["address"])
+    elif account["address"] == "0x0000000000000000000000000000000000000006":
+        print("pop account", account["address"])
+    elif account["address"] == "0x0000000000000000000000000000000000000002":
+        print("pop account", account["address"])
+    elif account["address"] == "0x0000000000000000000000000000000000000002":
+        print("pop account", account["address"])
+    elif account["address"] == "0x0000000000000000000000000000000000000008":
+        print("pop account", account["address"])
```
Comment (kingpinXD) [inline .github/actions/upgrade-testing/scripts/create_genesis.py:38]: Are these values constants for Athens 3? Woudl this still work if we want to use this script to export state from a different network

File: .github/actions/upgrade-testing/scripts/raise_gov_proposal.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import os
+import requests
+os.environ['NODE'] = "http://127.0.0.1:26657"
+CURRENT_HEIGHT = requests.get(f"{os.environ['NODE']}/status").json()["result"]["sync_info"]["latest_block_height"]
+UPGRADE_HEIGHT = int(CURRENT_HEIGHT) + (int(os.environ['PROPOSAL_TIME_SECONDS']) / int(os.environ['BLOCK_TIME_SECONDS'])) + 20
+github_file = open(os.environ["GITHUB_ENV"], "a+")
+github_file.write(f"UPGRADE_HEIGHT={UPGRADE_HEIGHT}")
+github_file.close()
+
+GOV_PROPOSAL = f"""zetacored tx gov submit-legacy-proposal software-upgrade "{os.environ['VERSION']}" \
```
Comment (kingpinXD) [inline .github/actions/upgrade-testing/scripts/raise_gov_proposal.py:10]: @gzukel  can we try moving away from the legacy proposal, and submit using the regular proposal instead?

**Review by CharlieMc0 — APPROVED** [review]

**Pull request merged**